TIVar.cpp

```diff
@@ -261,73 +275,99 @@ int TIVar::stringToStrVar8x(String s, uint8_t* strVar, enum Endpoint model) {
 		tokenlen++;
 	}
 
-	TIVar::intToSizeWord(tokenlen, strVar);
+	if (type == STR_89) {
+		strVar[0] = '\0';
```

Comment:
I unfortunately don't have a TI-89 to test on. But [according to the Link Guide](http://merthsoft.com/linkguide/ti89/vars.html#string), string variables are null terminated on the 89 series and do not have a length prefix.